cmd/aws-vpc-cni/main.go

```diff
@@ -164,15 +163,18 @@ type Range struct {
 	Gateway    net.IP      `json:"gateway,omitempty"`
 }
 
+// Wait for IPAMD health check to pass (maximum wait time is 60 seconds)
 func waitForIPAM() bool {
+	start := time.Now()
+	maxEnd := start.Add(time.Minute)
 	for {
 		cmd := exec.Command("./grpc-health-probe", "-addr", "127.0.0.1:50051", ">", "/dev/null", "2>&1")
-		var outb bytes.Buffer
-		cmd.Stdout = &outb
-		cmd.Run()
-		if outb.String() == "" {
+		if err := cmd.Run(); err == nil {
 			return true
 		}
+		if time.Now().After(maxEnd) {
```

Comment:
Yeah, I think we want this error so that the node does not get marked as ready. And while IPAMD will have the log, until we have an event for customers to look at, I think it is better for anyone running `kubectl log aws-node-xxxxx -n kube-system` to see this, which will point them to the IPAMD logs.

Having the `Run()` on one line is fine, I just split it up as I thought it was more readable

Reply:
Yeah makes sense about `grpc-health-probe` but sorry I meant do we need the 1 min check?

Reply:
Since we don't want to wait indefinitely, I figured one minute was a reasonable upper limit for IPAMD to initialize and start gRPC server. We could also increase connect and rpc timeout args to `grpc-health-probe` instead of looping here, but I figured that was more prone to error as I wasn't sure if an errant TCP reset from gRPC server was possible